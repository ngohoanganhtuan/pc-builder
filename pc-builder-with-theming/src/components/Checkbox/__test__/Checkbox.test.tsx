import { fireEvent, render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { Checkbox } from '@components';

describe('Test [Checkbox] component', () => {
  const handleCheckboxChanged = jest.fn();
  const initialProps = {
    id: 'checkbox-1',
    label: 'Checkbox label',
    value: 'checkbox-1',
    name: 'checkbox-1',
    checked: true,
    handleCheckboxChanged,
  };

  test('Component [Checkbox] matches DOM Snapshot', () => {
    const component = renderer.create(<Checkbox {...initialProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [Checkbox] should call handleCheckboxChanged function when user click the checkbox', () => {
    render(<Checkbox {...initialProps} />);
    fireEvent.click(screen.getByRole('checkbox'));
    expect(handleCheckboxChanged).toHaveBeenCalled();
  });
});
