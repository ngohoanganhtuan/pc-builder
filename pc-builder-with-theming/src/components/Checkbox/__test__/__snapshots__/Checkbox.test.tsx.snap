// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [Checkbox] component Component [Checkbox] matches DOM Snapshot 1`] = `
<div
  className="form-check"
>
  <input
    checked={true}
    className="form-check-input"
    id="checkbox-1"
    name="checkbox-1"
    onChange={[MockFunction]}
    type="checkbox"
    value="checkbox-1"
  />
  <label
    className="form-check-label brand-checkbox-label"
    htmlFor="checkbox-1"
  >
    Checkbox label
  </label>
</div>
`;
